beetsplug/convert.py

```diff
@@ -426,6 +482,21 @@ def convert_func(self, lib, opts, args):
         else:
             pretend = self.config['pretend'].get(bool)
 
+        if opts.link is not None:
+            link = opts.link
+
+            if opts.hardlink is not None:
+                hardlink = opts.hardlink
+            else:
+                hardlink = self.config['hardlink'].get(bool) and not link
+        else:
+            link = self.config['link'].get(bool)
+
+            if opts.hardlink is not None:
+                hardlink = opts.hardlink
+            else:
+                hardlink = self.config['hardlink'].get(bool)
```

Comment:
Maybe something like this would be less repetitive?

```py
if opts.link is not None:
    link = True
    hardlink = False
elif opts.hardlink is not None:
    link = False
    hardlink = True
else:
    # use config
````

Or maybe we could use an enum with three values instead of two separate flags?

Reply:
maybe using the `MoveOperation` could be a good idea